client/src/components/Map/UCIMap.js

```diff
@@ -45,6 +47,125 @@ export default class UCIMap extends PureComponent {
+                            if (waypoint != 0) {
+                                poly.push(<Polyline color={colors[waypoint - 1]} positions={path[waypoint]} />);
+                                poly.push(
+                                    <Polyline
+                                        color={colors[waypoint - 1]}
+                                        positions={[path[waypoint][path[waypoint].length - 1], coords_array[waypoint]]}
+                                        dashArray="4"
+                                    />
+                                );
+                                poly.push(
+                                    <Marker
+                                        position={path[waypoint][Math.floor(path[waypoint].length / 2)]}
+                                        icon={Leaflet.divIcon({
+                                            iconAnchor: [0, 14],
+                                            labelAnchor: [-3.5, 0],
+                                            popupAnchor: [0, -21],
+                                            className: '',
+                                            iconSize: [1000, 14],
+                                            html: `<div style="background-color: white;border-left-color: ${
+                                                colors[waypoint - 1]
+                                            };border-left-style: solid;width: fit-content;border-left-width: 5px;padding-left: 10px;padding-right: 10px;padding-top: 4px;padding-bottom: 4px;">
+                                    <span style="color:${colors[waypoint - 1]}"> ${
+                                                obj['routes'][0]['legs'][waypoint - 1]['duration'] > 30
+                                                    ? Math.round(
+                                                          obj['routes'][0]['legs'][waypoint - 1]['duration'] / 60
+                                                      ).toString() + ' min'
+                                                    : '<1 min'
+                                            } </span>
+                                    <br>
+                                    <span style="color:#888888">
+                                                ${
+                                                    (
+                                                        Math.floor(
+                                                            obj['routes'][0]['legs'][waypoint - 1]['distance'] /
+                                                                1.609 /
+                                                                10
+                                                        ) / 100
+                                                    ).toString() + ' mi'
+                                                }
+                                                </span>
+                                            </div>`,
+                                        })}
```

Comment:
The math here is brilliant, but it's hard to follow inside of the string interpolation.
Pull all the computations out of the string and set them as variables above line 124 (`poly.push(`)